src/main/java/Eelyon/Eelyon.java

```diff
@@ -0,0 +1,296 @@
+    static final String FILE_PATH = "src/main/java/Eelyon/data/data.txt";
+
+    public static void printList() {
+        System.out.println(LINE);
+        if (list.isEmpty()) {
+            System.out.println("List is empty");
+        } else {
+            for (Task task : list) {
+                System.out.print(list.indexOf(task) + 1 + ".");
+                System.out.println(task);
+            }
+        }
+        System.out.println(LINE);
+    }
+
+    public static void setMarked(int index) {
+        if (index >= list.size() || index < 0) {
+            System.out.println(LINE);
+            System.out.println("Invalid index");
+            System.out.println(LINE);
+            return;
+        } else {
+            list.get(index).setDone(true);
+            System.out.println(LINE);
+            System.out.println("Nice! I've marked this task as done:\n" + "[" + list.get(index).getStatusIcon() + "] " + list.get(index).getDescription());
+            System.out.println(LINE);
+        }
+    }
+
+    public static void setUnmarked(int index) {
+        if (index >= list.size() || index < 0) {
+            System.out.println(LINE);
+            System.out.println("Invalid index");
+            System.out.println(LINE);
+        } else {
+            list.get(index).setDone(false);
+            System.out.println(LINE);
+            System.out.println("Ok, I've marked this task as not done yet:\n" + "[" + list.get(index).getStatusIcon() + "] " + list.get(index).getDescription());
+            System.out.println(LINE);
+        }
+    }
+
+    public static void deleteTask(int index) {
+        if (index >= list.size() || index < 0) {
+            System.out.println(LINE);
+            System.out.println("Invalid index");
+            System.out.println(LINE);
+        } else {
+            System.out.println(LINE);
+            System.out.println("Noted. Removing this task:\n" + list.get(index));
+            list.remove(index);
+            System.out.printf("Now you have %d tasks in the list:\n", list.size());
+            System.out.println(LINE);
+        }
+    }
+
+    public static void addTask(Task task) {
+        list.add(task);
+        System.out.println(LINE + "added: " + task.getDescription() + "\n");
+        System.out.printf("You now have %d tasks in the list\n", list.size());
+        System.out.println(LINE);
+    }
+
+    public static void checkDescription(CommandType commandType, String input) {
+        if (commandType.getCommandType().equals("todo")) {
+            if (input.substring(input.indexOf("todo") + "todo".length()).trim().length() <= 0) {
+                throw new EmptyDescriptionException("No empty descriptions allowed dawg. Try again.");
+            }
+        } else if (input.substring(input.indexOf(commandType.getCommandType()) + commandType.getCommandType().length(), input.indexOf(commandType.getEndCommand())).trim().length() <= 0) {
```

Comment:
Lines with >120 chars can hinder readability. The statement is also very complex. Consider breaking it into parts. Else wrap the text onto the next line. Although in this case I would suggest you make this expression less complex.

Reply:
This applies to lines 103, 126, 132 too.

Reply:
Notes on [avoiding complex expressions](https://nus-cs2113-ay2425s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-complicated-expressions)